integration-testing/onboard_test.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2024 Illumio, Inc. All Rights Reserved.
+
+package integrationtesting
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/spf13/viper"
+)
+
+// Cluster represents a Kubernetes cluster with relevant metadata.
+type Cluster struct {
+	ID            string `json:"id"`
+	IllumioRegion string `json:"illumio_region"`
+	Enabled       bool   `json:"enabled"`
+	Onboarded     bool   `json:"onboarded"`
+}
+
+// ClustersResponse represents the response containing cluster data.
+type ClustersResponse struct {
+	Clusters      []Cluster `json:"clusters"`
+	TotalSize     int       `json:"total_size"`
+	NextPageToken string    `json:"next_page_token"`
+	PrevPageToken string    `json:"prev_page_token"`
+	Page          int       `json:"page"`
+}
+
+type OffboardRequest struct {
+	ClusterIds []string `json:"cluster_ids"`
+}
+
+// Config holds configuration values needed for the API requests.
+type Config struct {
+	TenantId     string
+	CloudIdKey   string
+	CloudIdValue string
+	UserId       string
+}
+
+// NewHTTPClient returns a reusable HTTP client.
+func NewHTTPClient() *http.Client {
+	return &http.Client{}
+}
+
+// getAuthorizationHeader returns the Basic Authorization header using the provided credentials.
+func getAuthorizationHeader(key, secret string) string {
+	credentials := fmt.Sprintf("%s:%s", key, secret)
+	encodedCredentials := base64.StdEncoding.EncodeToString([]byte(credentials))
+	return fmt.Sprintf("Basic %s", encodedCredentials)
+}
+
+// buildRequest creates a new HTTP request with common headers.
+func buildRequest(method, url string, body io.Reader, config Config) (*http.Request, error) {
```

Comment:
NIt. I think "new" is more idiomatic.
```suggestion
func newRequest(method, url string, body io.Reader, config Config) (*http.Request, error) {
```